{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\tqdm\\auto.py:22: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "from scikit_stan import GLM\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.pipeline import Pipeline\n",
    "from sklearn import datasets \n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.model_selection import GridSearchCV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "X, y, true_weights = datasets.make_regression(\n",
    "    n_samples=100,\n",
    "    n_features=100,\n",
    "    n_informative=10,\n",
    "    noise=8,\n",
    "    coef=True,\n",
    "    random_state=42,\n",
    ")\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y,\n",
    "                                                     random_state=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py:406: UserWarning: \n",
      "                    Link function not specified. Using default link function 'identity'\n",
      "                    for family 'gaussian'.\n",
      "                \n",
      "  \"\"\"\n",
      "c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py:500: UserWarning: Prior on intercept not specified. Using default prior.\n",
      "                alpha ~ normal(mu(y), 2.5 * sd(y)) if Gaussian family else normal(0, 2.5)\n",
      "  alpha ~ normal(mu(y), 2.5 * sd(y)) if Gaussian family else normal(0, 2.5)\"\"\"\n",
      "c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py:546: UserWarning: Prior on auxiliary parameter not specified. Using default unscaled prior\n",
      "                        sigma ~ exponential(1)\n",
      "                    \n",
      "  \"\"\"\n",
      "15:54:55 - cmdstanpy - INFO - CmdStan start processing\n",
      "chain 1 |\u001b[33m          \u001b[0m| 00:00 Status\n",
      "\u001b[A\n",
      "\n",
      "chain 1 |\u001b[33m▍         \u001b[0m| 00:00 Status\n",
      "\u001b[A\n",
      "\n",
      "chain 1 |\u001b[33m▉         \u001b[0m| 00:00 Iteration:    1 / 2000 [  0%]  (Warmup)\n",
      "\u001b[A\n",
      "\n",
      "chain 1 |\u001b[33m████      \u001b[0m| 00:00 Iteration:  700 / 2000 [ 35%]  (Warmup)\n",
      "\u001b[A\n",
      "\n",
      "chain 1 |\u001b[34m█████▉    \u001b[0m| 00:00 Iteration: 1001 / 2000 [ 50%]  (Sampling)\n",
      "\u001b[A\n",
      "\n",
      "chain 1 |\u001b[34m███████▋  \u001b[0m| 00:00 Iteration: 1400 / 2000 [ 70%]  (Sampling)\n",
      "\u001b[A\n",
      "\n",
      "\u001b[A\u001b[A\n",
      "chain 1 |\u001b[34m█████████ \u001b[0m| 00:01 Iteration: 1700 / 2000 [ 85%]  (Sampling)\n",
      "\n",
      "chain 1 |\u001b[34m██████████\u001b[0m| 00:01 Sampling completed                       \n",
      "chain 2 |\u001b[34m██████████\u001b[0m| 00:01 Sampling completed                       \n",
      "chain 3 |\u001b[34m██████████\u001b[0m| 00:01 Sampling completed                       \n",
      "chain 4 |\u001b[34m██████████\u001b[0m| 00:01 Sampling completed                       "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                                                                                                                                                                                                                                                                                                "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "15:54:56 - cmdstanpy - INFO - CmdStan done processing.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "15:54:56 - cmdstanpy - INFO - Chain [1] start processing\n",
      "15:54:56 - cmdstanpy - INFO - Chain [2] start processing\n",
      "15:54:56 - cmdstanpy - INFO - Chain [3] start processing\n",
      "15:54:56 - cmdstanpy - INFO - Chain [4] start processing\n",
      "15:54:58 - cmdstanpy - INFO - Chain [2] done processing\n",
      "15:54:58 - cmdstanpy - INFO - Chain [3] done processing\n",
      "15:54:58 - cmdstanpy - INFO - Chain [4] done processing\n",
      "15:54:58 - cmdstanpy - INFO - Chain [1] done processing\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.9911734663273118"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pipe = Pipeline([('scaler', StandardScaler()), ('glm', GLM( seed=1234))])\n",
    "pipe.fit(X_train, y_train)\n",
    "pipe.score(X_test, y_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py:406: UserWarning: \n",
      "                    Link function not specified. Using default link function 'identity'\n",
      "                    for family 'gaussian'.\n",
      "                \n",
      "  \"\"\"\n",
      "c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py:500: UserWarning: Prior on intercept not specified. Using default prior.\n",
      "                alpha ~ normal(mu(y), 2.5 * sd(y)) if Gaussian family else normal(0, 2.5)\n",
      "  alpha ~ normal(mu(y), 2.5 * sd(y)) if Gaussian family else normal(0, 2.5)\"\"\"\n",
      "c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py:546: UserWarning: Prior on auxiliary parameter not specified. Using default unscaled prior\n",
      "                        sigma ~ exponential(1)\n",
      "                    \n",
      "  \"\"\"\n",
      "15:54:58 - cmdstanpy - INFO - CmdStan start processing\n",
      "chain 1 |\u001b[33m          \u001b[0m| 00:00 Status\n",
      "\u001b[A\n",
      "\n",
      "chain 1 |\u001b[33m▉         \u001b[0m| 00:00 Iteration:    1 / 2000 [  0%]  (Warmup)\n",
      "\n",
      "\u001b[A\u001b[A\n",
      "chain 1 |\u001b[33m████      \u001b[0m| 00:00 Iteration:  700 / 2000 [ 35%]  (Warmup)\n",
      "\n",
      "\u001b[A\u001b[A\n",
      "\u001b[A\n",
      "chain 1 |\u001b[34m██████▎   \u001b[0m| 00:00 Iteration: 1100 / 2000 [ 55%]  (Sampling)\n",
      "\n",
      "\u001b[A\u001b[A\n",
      "chain 1 |\u001b[34m████████▏ \u001b[0m| 00:00 Iteration: 1500 / 2000 [ 75%]  (Sampling)\n",
      "\n",
      "\u001b[A\u001b[A\n",
      "chain 1 |\u001b[34m█████████▌\u001b[0m| 00:01 Iteration: 1800 / 2000 [ 90%]  (Sampling)\n",
      "\n",
      "chain 1 |\u001b[34m██████████\u001b[0m| 00:01 Sampling completed                       \n",
      "chain 2 |\u001b[34m██████████\u001b[0m| 00:01 Sampling completed                       \n",
      "chain 3 |\u001b[34m██████████\u001b[0m| 00:01 Sampling completed                       \n",
      "chain 4 |\u001b[34m██████████\u001b[0m| 00:01 Sampling completed                       "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                                                                                                                                                                                                                                                                                                "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "15:54:59 - cmdstanpy - INFO - CmdStan done processing.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "15:55:00 - cmdstanpy - INFO - Chain [1] start processing\n",
      "15:55:00 - cmdstanpy - INFO - Chain [2] start processing\n",
      "15:55:00 - cmdstanpy - INFO - Chain [3] start processing\n",
      "15:55:00 - cmdstanpy - INFO - Chain [4] start processing\n",
      "15:55:01 - cmdstanpy - INFO - Chain [1] done processing\n",
      "15:55:01 - cmdstanpy - INFO - Chain [4] done processing\n",
      "15:55:01 - cmdstanpy - INFO - Chain [3] done processing\n",
      "15:55:01 - cmdstanpy - INFO - Chain [2] done processing\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.9899535381342051"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "glm_unscaled = GLM(seed=1234) \n",
    "glm_unscaled.fit(X_train, y_train)\n",
    "glm_unscaled.score(X_test, y_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'algorithm': 'sample', 'algorithm_params': None, 'autoscale': False, 'family': 'bernoulli', 'link': 'logit', 'prior_aux': None, 'prior_intercept': None, 'priors': None, 'seed': 1234}\n"
     ]
    }
   ],
   "source": [
    "pca = PCA()\n",
    "scaler = StandardScaler()\n",
    "logisticregression = GLM(family=\"bernoulli\", link=\"logit\", seed=1234)\n",
    "\n",
    "pipeline = Pipeline([('pca', pca), ('scaler', scaler), ('log_reg', logisticregression)])\n",
    "print(logisticregression.get_params())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:372: FitFailedWarning: \n",
      "25 fits failed out of a total of 25.\n",
      "The score on these train-test partitions for these parameters will be set to nan.\n",
      "If these failures are not expected, you can try to debug them by setting error_score='raise'.\n",
      "\n",
      "Below are more details about the failures:\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\dlpj888y.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discrete7kso74x6\\\\glm_v_discrete-20220808155505_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discrete7kso74x6\\glm_v_discrete-20220808155505_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discrete7kso74x6\\glm_v_discrete-20220808155505_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\xe_dcimp.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discrete9jwfmke4\\\\glm_v_discrete-20220808155505_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discrete9jwfmke4\\glm_v_discrete-20220808155505_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discrete9jwfmke4\\glm_v_discrete-20220808155505_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\gt0mla_f.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discrete86ml0emh\\\\glm_v_discrete-20220808155505_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discrete86ml0emh\\glm_v_discrete-20220808155505_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discrete86ml0emh\\glm_v_discrete-20220808155505_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\px4st6od.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discretek1xg1c1n\\\\glm_v_discrete-20220808155505_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discretek1xg1c1n\\glm_v_discrete-20220808155505_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discretek1xg1c1n\\glm_v_discrete-20220808155505_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\e3hy5c2y.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discreteuap_2b9s\\\\glm_v_discrete-20220808155505_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discreteuap_2b9s\\glm_v_discrete-20220808155505_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discreteuap_2b9s\\glm_v_discrete-20220808155505_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\6uwpdlfm.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discretealuiysoo\\\\glm_v_discrete-20220808155506_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discretealuiysoo\\glm_v_discrete-20220808155506_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discretealuiysoo\\glm_v_discrete-20220808155506_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\w3_xd5hj.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discretefajpkohv\\\\glm_v_discrete-20220808155506_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discretefajpkohv\\glm_v_discrete-20220808155506_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discretefajpkohv\\glm_v_discrete-20220808155506_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\i7xupz_y.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discreteavc2818u\\\\glm_v_discrete-20220808155506_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discreteavc2818u\\glm_v_discrete-20220808155506_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discreteavc2818u\\glm_v_discrete-20220808155506_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\8fxqfk6q.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discreteusz8jmfy\\\\glm_v_discrete-20220808155506_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discreteusz8jmfy\\glm_v_discrete-20220808155506_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discreteusz8jmfy\\glm_v_discrete-20220808155506_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\tbnnrnoh.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discrete6_3zwvja\\\\glm_v_discrete-20220808155506_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discrete6_3zwvja\\glm_v_discrete-20220808155506_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discrete6_3zwvja\\glm_v_discrete-20220808155506_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\kex82ccr.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discreteb6afvr3k\\\\glm_v_discrete-20220808155506_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discreteb6afvr3k\\glm_v_discrete-20220808155506_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discreteb6afvr3k\\glm_v_discrete-20220808155506_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\n2zcncu1.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discreteklnrvh11\\\\glm_v_discrete-20220808155506_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discreteklnrvh11\\glm_v_discrete-20220808155506_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discreteklnrvh11\\glm_v_discrete-20220808155506_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\ng9ks02p.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discrete3x6hdhaq\\\\glm_v_discrete-20220808155506_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discrete3x6hdhaq\\glm_v_discrete-20220808155506_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discrete3x6hdhaq\\glm_v_discrete-20220808155506_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\u5mwyp19.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discrete44x6clyt\\\\glm_v_discrete-20220808155506_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discrete44x6clyt\\glm_v_discrete-20220808155506_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discrete44x6clyt\\glm_v_discrete-20220808155506_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\6000zaa6.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discretegowsav1d\\\\glm_v_discrete-20220808155507_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discretegowsav1d\\glm_v_discrete-20220808155507_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discretegowsav1d\\glm_v_discrete-20220808155507_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\hsfjbtle.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discretejqzv5q49\\\\glm_v_discrete-20220808155507_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discretejqzv5q49\\glm_v_discrete-20220808155507_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discretejqzv5q49\\glm_v_discrete-20220808155507_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\s00h23mw.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discreteibboneiy\\\\glm_v_discrete-20220808155507_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discreteibboneiy\\glm_v_discrete-20220808155507_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discreteibboneiy\\glm_v_discrete-20220808155507_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\__crgi22.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discrete5id3zlmh\\\\glm_v_discrete-20220808155507_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discrete5id3zlmh\\glm_v_discrete-20220808155507_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discrete5id3zlmh\\glm_v_discrete-20220808155507_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\mrlp_nds.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discretedgujplyu\\\\glm_v_discrete-20220808155507_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discretedgujplyu\\glm_v_discrete-20220808155507_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discretedgujplyu\\glm_v_discrete-20220808155507_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\ukxghinl.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discretedrcou00u\\\\glm_v_discrete-20220808155507_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discretedrcou00u\\glm_v_discrete-20220808155507_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discretedrcou00u\\glm_v_discrete-20220808155507_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\esf37bjz.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discretefuf1xa1x\\\\glm_v_discrete-20220808155507_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discretefuf1xa1x\\glm_v_discrete-20220808155507_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discretefuf1xa1x\\glm_v_discrete-20220808155507_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\dcytdhz7.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discretequ_1ofki\\\\glm_v_discrete-20220808155508_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discretequ_1ofki\\glm_v_discrete-20220808155508_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discretequ_1ofki\\glm_v_discrete-20220808155508_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\8wgmavjp.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discretel08_bmut\\\\glm_v_discrete-20220808155508_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discretel08_bmut\\glm_v_discrete-20220808155508_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discretel08_bmut\\glm_v_discrete-20220808155508_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\wv8j1_l8.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmpmyjuws8q\\\\glm_v_discrete6fb_ui6e\\\\glm_v_discrete-20220808155508_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discrete6fb_ui6e\\glm_v_discrete-20220808155508_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmpmyjuws8q\\glm_v_discrete6fb_ui6e\\glm_v_discrete-20220808155508_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 680, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\", line 394, in fit\n",
      "    self._final_estimator.fit(Xt, y, **fit_params_last_step)\n",
      "  File \"c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\", line 587, in fit\n",
      "    **self.algorithm_params if self.algorithm_params else {},\n",
      "  File \"c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\", line 1188, in sample\n",
      "    raise RuntimeError(msg)\n",
      "RuntimeError: Error during sampling:\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Exception: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\n",
      "Command and output files:\n",
      "RunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n",
      " cmd (chain 1):\n",
      "\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\9e99m43p.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp75wiwv0q\\\\glm_v_discrete_q60hhvi\\\\glm_v_discrete-20220808155508_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n",
      " retcodes=[1, 1, 1, 1]\n",
      " per-chain output files (showing chain 1 only):\n",
      " csv_file:\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discrete_q60hhvi\\glm_v_discrete-20220808155508_1.csv\n",
      " console_msgs (if any):\n",
      "\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp75wiwv0q\\glm_v_discrete_q60hhvi\\glm_v_discrete-20220808155508_0-stdout.txt\n",
      "Consider re-running with show_console=True if the above output is unclear!\n",
      "\n",
      "  warnings.warn(some_fits_failed_message, FitFailedWarning)\n",
      "c:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_search.py:972: UserWarning: One or more of the test scores are non-finite: [nan nan nan nan nan]\n",
      "  category=UserWarning,\n",
      "c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py:500: UserWarning: Prior on intercept not specified. Using default prior.\n",
      "                alpha ~ normal(mu(y), 2.5 * sd(y)) if Gaussian family else normal(0, 2.5)\n",
      "  alpha ~ normal(mu(y), 2.5 * sd(y)) if Gaussian family else normal(0, 2.5)\"\"\"\n",
      "c:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py:546: UserWarning: Prior on auxiliary parameter not specified. Using default unscaled prior\n",
      "                        sigma ~ exponential(1)\n",
      "                    \n",
      "  \"\"\"\n",
      "15:55:08 - cmdstanpy - INFO - CmdStan start processing\n",
      "chain 1 |\u001b[33m          \u001b[0m| 00:00 Status\n",
      "\u001b[A\n",
      "\n",
      "\u001b[A\u001b[A15:55:08 - cmdstanpy - ERROR - Chain [1] error: error during processing Operation not permitted\n",
      "15:55:08 - cmdstanpy - ERROR - Chain [2] error: error during processing Operation not permitted\n",
      "15:55:08 - cmdstanpy - ERROR - Chain [3] error: error during processing Operation not permitted\n",
      "15:55:08 - cmdstanpy - ERROR - Chain [4] error: error during processing Operation not permitted\n",
      "chain 1 |\u001b[33m██████████\u001b[0m| 00:00 Sampling completed\n",
      "chain 2 |\u001b[33m██████████\u001b[0m| 00:00 Sampling completed\n",
      "\n",
      "chain 3 |\u001b[33m██████████\u001b[0m| 00:00 Sampling completed\n",
      "\n",
      "\n",
      "chain 4 |\u001b[33m██████████\u001b[0m| 00:00 Sampling completed"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                                                                                                                                                                                                                                                                                                "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "15:55:08 - cmdstanpy - INFO - CmdStan done processing.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "ename": "RuntimeError",
     "evalue": "Error during sampling:\nException: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\nException: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\nException: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\nException: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\nCommand and output files:\nRunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n cmd (chain 1):\n\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp_u71echc\\\\4siw0y7t.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp_u71echc\\\\glm_v_discreter1jfdm_9\\\\glm_v_discrete-20220808155508_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n retcodes=[1, 1, 1, 1]\n per-chain output files (showing chain 1 only):\n csv_file:\n\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp_u71echc\\glm_v_discreter1jfdm_9\\glm_v_discrete-20220808155508_1.csv\n console_msgs (if any):\n\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp_u71echc\\glm_v_discreter1jfdm_9\\glm_v_discrete-20220808155508_0-stdout.txt\nConsider re-running with show_console=True if the above output is unclear!",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_16828\\706522518.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      5\u001b[0m }\n\u001b[0;32m      6\u001b[0m \u001b[0msearchgrid\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mGridSearchCV\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpipeline\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mparam_grid\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mn_jobs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m2\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 7\u001b[1;33m \u001b[0msearchgrid\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mx_digits\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0my_digits\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      8\u001b[0m \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"Best parameter (CV score=%0.3f):\"\u001b[0m \u001b[1;33m%\u001b[0m \u001b[0msearchgrid\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbest_score_\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      9\u001b[0m \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msearchgrid\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbest_params_\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\model_selection\\_search.py\u001b[0m in \u001b[0;36mfit\u001b[1;34m(self, X, y, groups, **fit_params)\u001b[0m\n\u001b[0;32m    924\u001b[0m             \u001b[0mrefit_start_time\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    925\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0my\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 926\u001b[1;33m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbest_estimator_\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mX\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0my\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mfit_params\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    927\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    928\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbest_estimator_\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mX\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mfit_params\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\sklearn\\pipeline.py\u001b[0m in \u001b[0;36mfit\u001b[1;34m(self, X, y, **fit_params)\u001b[0m\n\u001b[0;32m    392\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_final_estimator\u001b[0m \u001b[1;33m!=\u001b[0m \u001b[1;34m\"passthrough\"\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    393\u001b[0m                 \u001b[0mfit_params_last_step\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mfit_params_steps\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msteps\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;33m-\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 394\u001b[1;33m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_final_estimator\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mXt\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0my\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mfit_params_last_step\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    395\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    396\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\users\\alizm\\documents\\flatironinstitute\\scikit-stan\\scikit_stan\\generalized_linear_regression\\glm.py\u001b[0m in \u001b[0;36mfit\u001b[1;34m(self, X, y, show_console)\u001b[0m\n\u001b[0;32m    585\u001b[0m             \u001b[0mseed\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mseed_\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    586\u001b[0m             \u001b[0msig_figs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m9\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 587\u001b[1;33m             \u001b[1;33m**\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0malgorithm_params\u001b[0m \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0malgorithm_params\u001b[0m \u001b[1;32melse\u001b[0m \u001b[1;33m{\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    588\u001b[0m         )\n\u001b[0;32m    589\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\Users\\alizm\\anaconda3\\envs\\intern\\lib\\site-packages\\cmdstanpy\\model.py\u001b[0m in \u001b[0;36msample\u001b[1;34m(self, data, chains, parallel_chains, threads_per_chain, seed, chain_ids, inits, iter_warmup, iter_sampling, save_warmup, thin, max_treedepth, metric, step_size, adapt_engaged, adapt_delta, adapt_init_phase, adapt_metric_window, adapt_step_size, fixed_param, output_dir, sig_figs, save_latent_dynamics, save_profile, show_progress, show_console, refresh, time_fmt, force_one_process_per_chain)\u001b[0m\n\u001b[0;32m   1186\u001b[0m                     \u001b[1;33m+\u001b[0m \u001b[1;34m' output is unclear!'\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1187\u001b[0m                 )\n\u001b[1;32m-> 1188\u001b[1;33m                 \u001b[1;32mraise\u001b[0m \u001b[0mRuntimeError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmsg\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1189\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0merrors\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1190\u001b[0m                 msg = (\n",
      "\u001b[1;31mRuntimeError\u001b[0m: Error during sampling:\nException: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\nException: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\nException: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\nException: glm_v_discrete_model_namespace::glm_v_discrete_model: family is 6, but must be less than or equal to 5.000000 (in 'C:/Users/alizm/Documents/FlatironInstitute/scikit-stan/scikit_stan/stan_files/glm_v_discrete.stan', line 15, column 2 to column 31)\nCommand and output files:\nRunSet: chains=4, chain_ids=[1, 2, 3, 4], num_processes=4\n cmd (chain 1):\n\t['c:\\\\users\\\\alizm\\\\documents\\\\flatironinstitute\\\\scikit-stan\\\\scikit_stan\\\\stan_files\\\\glm_v_discrete.exe', 'id=1', 'random', 'seed=1234', 'data', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp_u71echc\\\\4siw0y7t.json', 'output', 'file=C:\\\\Users\\\\alizm\\\\AppData\\\\Local\\\\Temp\\\\tmp_u71echc\\\\glm_v_discreter1jfdm_9\\\\glm_v_discrete-20220808155508_1.csv', 'sig_figs=9', 'method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']\n retcodes=[1, 1, 1, 1]\n per-chain output files (showing chain 1 only):\n csv_file:\n\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp_u71echc\\glm_v_discreter1jfdm_9\\glm_v_discrete-20220808155508_1.csv\n console_msgs (if any):\n\tC:\\Users\\alizm\\AppData\\Local\\Temp\\tmp_u71echc\\glm_v_discreter1jfdm_9\\glm_v_discrete-20220808155508_0-stdout.txt\nConsider re-running with show_console=True if the above output is unclear!"
     ]
    }
   ],
   "source": [
    "\n",
    "x_digits, y_digits = datasets.load_digits(return_X_y=True)\n",
    "param_grid = {\n",
    "    \"pca__n_components\": [5, 15, 30, 45, 60],\n",
    "    #\"logistic__C\": np.logspace(-4, 4, 4), # this should be a specific parameter of the GLM! \n",
    "}\n",
    "searchgrid = GridSearchCV(pipeline, param_grid, n_jobs=2)\n",
    "searchgrid.fit(x_digits, y_digits)\n",
    "print(\"Best parameter (CV score=%0.3f):\" % searchgrid.best_score_)\n",
    "print(searchgrid.best_params_)\n",
    "pca.fit(x_digits)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.7.12 ('intern')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.13"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "160f8a19148fea9a7cf8ea4744d98c012bbc4693e4d32de3466f175967d43183"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
